apps/src/templates/sectionsRefresh/SectionsSetUp.jsx

```diff
@@ -0,0 +1,56 @@
+import React, {useState} from 'react';
+import i18n from '@cdo/locale';
+import SetUpSectionForm from './SetUpSectionForm';
+import Button from '@cdo/apps/templates/Button';
+import moduleStyles from './sections-refresh.module.scss';
+
+const useSections = () => {
+  const [sections, setSections] = useState([{}]);
+
+  const updateSection = (sectionIdx, keyToUpdate, val) => {
+    const newSections = sections.map((section, idx) => {
+      if (idx === sectionIdx) {
+        return {
+          ...section,
+          [keyToUpdate]: val
+        };
+      } else {
+        return section;
+      }
+    });
+    setSections(newSections);
```

Comment:
I'm having a hard time wrapping my brain around the plural vs singular here. Does this function update a single section, or several at once? Or maybe I don't understand the name 'useSections'. Could you maybe just add comments describing what the pieces do?

Reply:
`setSections` updates the entire list of sections but `updateSection` only updates one. I'll add some comments here -- this is a custom hook pattern I've used in the past to hide some of the details you're asking about :)